### PR TITLE
PRT-142: Clean up contracts interfaces

### PR DESCRIPTION
**Pull request by lazaralex98**
Convert the Toucan contracts in the "CO2KEN_contracts" folder to .sol interfaces and add the .sol interfaces that I previously created for the tokenizer repo.

### CODE REVIEW

File: contracts/interfaces/IRetirementCertificates.sol [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/StringsUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import "./IToucanContractRegistry.sol";
+
+interface RetirementCertificates is IERC721Upgradeable {
+    function version() external returns (string memory);
```
Comment (kargakis) [inline contracts/interfaces/IRetirementCertificates.sol:17]: Why do you need to read the `version` of the contract?

File: test/Others.ts [hunk 2]
```diff
@@ -2,30 +2,22 @@ import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { expect } from "chai";
```
Comment (kargakis) [inline test/Others.ts:2]: Do these tests work now? Last time I run them, they weren't working.

Comment (lazaralex98) [reply test/Others.ts:2]: I tested before pushing this time and they worked.  I remember they worked (for me) last time too.

File: contracts/interfaces/IRetirementCertificates.sol (hunk 1 above)
Comment (lazaralex98) [reply contracts/interfaces/IRetirementCertificates.sol:17]: It's a public method, maybe people could be curious or could have some sort of edge case for it. I don't really see a drawback to keeping it.

File: test/Others.ts (hunk 2 above)
Comment (kargakis) [reply test/Others.ts:2]: Can you please enable them in CI then? Update the Github workflow to uncomment the part I have commented out.

**Comment by lazaralex98** [comment]
Could this be some sort of env issue? Any quick ideas that might save me some time?

```
An unexpected error occurred:
[115](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:115)

[116](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:116)
TypeError [ERR_INVALID_URL]: Invalid URL
[117](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:117)
    at new NodeError (node:internal/errors:371:5)
[118](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:118)
    at onParseError (node:internal/url:552:9)
[119](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:119)
    at new URL (node:internal/url:628:5)
[120](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:120)
    at new HttpProvider (/home/runner/work/example-implementations/example-implementations/node_modules/hardhat/src/internal/core/providers/http.ts:47:17)
[121](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:121)
    at makeForkClient (/home/runner/work/example-implementations/example-implementations/node_modules/hardhat/src/internal/hardhat-network/provider/utils/makeForkClient.ts:34:20)
[122](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:122)
    at Function.create (/home/runner/work/example-implementations/example-implementations/node_modules/hardhat/src/internal/hardhat-network/provider/node.ts:161:31)
[123](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:123)
    at HardhatNetworkProvider._init (/home/runner/work/example-implementations/example-implementations/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:251:46)
[124](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:124)
    at async HardhatNetworkProvider._send (/home/runner/work/example-implementations/example-implementations/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:192:5)
[125](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:125)
    at async HardhatNetworkProvider.request (/home/runner/work/example-implementations/example-implementations/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:118:18)
[126](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:126)
    at async SimpleTaskDefinition.action (/home/runner/work/example-implementations/example-implementations/node_modules/hardhat/src/builtin-tasks/test.ts:167:37) {
[127](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:127)
  input: '',
[128](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:128)
  code: 'ERR_INVALID_URL'
[129](https://github.com/ToucanProtocol/example-implementations/runs/5868535260?check_suite_focus=true#step:6:129)
}
```

File: contracts/interfaces/IRetirementCertificates.sol (hunk 1 above)
Comment (marceljay) [reply contracts/interfaces/IRetirementCertificates.sol:17]: Good question. Would be interesting in learning best practices for what to include in interfaces. 
The question is, why would another contract want to call this function? (but there are possible use cases...)

Comment (lazaralex98) [reply contracts/interfaces/IRetirementCertificates.sol:17]: If you guys really want that gone from the interface, I'm not against it 🤷🏻‍♂️.

Comment (kargakis) [reply contracts/interfaces/IRetirementCertificates.sol:17]: I'd be in favor of removing until proven that it's needed. It sounds like a not so obvious use case to need this function so whoever has it will most likely know how to update their interfaces. :)

**Comment by lazaralex98** [comment]
I can't figure out why the build issue is happening. I think it's a good idea to turn tests back off, push this so interfaces are available in a clean way to outside devs, and open a backlog issue to figure this build/test problem out later. cc @kargakis  @marceljay

**Comment by kargakis** [comment]
> I can't figure out why the build issue is happening. I think it's a good idea to turn tests back off, push this so interfaces are available in a clean way to outside devs, and open a backlog issue to figure this build/test problem out later. cc @kargakis @marceljay

Sounds good, please create a follow-up ticket to track the fix.

File: contracts/interfaces/IRetirementCertificates.sol (hunk 1 above)
Comment (lazaralex98) [reply contracts/interfaces/IRetirementCertificates.sol:17]: Removed it will be then
